bin/scpipe/lib/sc/pipe.rb

```diff
@@ -69,37 +71,32 @@ def prepare_pipe
       end
 
       def run_pipe
-        @@pipeproc = Proc.new {
-          trap("INT") do
-            Process.exit
-          end
-          rundir = Dir.pwd
+        rundir = Dir.pwd
```

Comment:
The contents of this while loop were formerly executed in a seperate process, which was killed and recreated on receiving INT or HUP signals. This probably has always lead to some kind of process buildup in various edge cases.